ddtrace/tracer/option.go

```diff
@@ -82,10 +83,8 @@ type StartOption func(*config)
 
 // defaults sets the default values for a config.
 func defaults(c *config) {
-	c.serviceName = filepath.Base(os.Args[0])
 	c.sampler = NewAllSampler()
 	c.agentAddr = defaultAddress
-
```

Comment:
Please add this one back. It does separate two pieces of code doing different things.